frontend_server/lib/server.dart

```diff
@@ -231,42 +221,49 @@ class _FrontendCompiler implements CompilerInterface {
   Uri _ensureFolderPath(String path) {
     // This is a URI, not a file path, so the forward slash is correct even
     // on Windows.
-    if (!path.endsWith('/'))
+    if (!path.endsWith('/')) {
       path = '$path/';
+    }
     return Uri.base.resolve(path);
   }
 
-  static String _severityName(Severity severity) {
-    switch (severity) {
-      case Severity.error:
-        return "Error";
-      case Severity.internalProblem:
-        return "Internal problem";
-      case Severity.nit:
-        return "Nit";
-      case Severity.warning:
-        return "Warning";
-      default:
-        return severity.toString();
-    }
+  /// Runs the given function [f] in a Zone that redirects all prints into
+  /// [_outputStream].
+  Future<T> _runWithPrintRedirection<T>(Future<T> f()) {
+    return runZoned(() => new Future<T>(f),
+        zoneSpecification: new ZoneSpecification(
+            print: (Zone self, ZoneDelegate parent, Zone zone, String line) =>
+                _outputStream.writeln(line)));
```

Comment:
This is an example of where dartfmt's output makes the code nigh-on unreadable. To figure out the structure here requires very close reading. I recommend sprinkling some trailing commas and rerunning dartfmt to get it to behave better. I'll add it to our list of dartfmt issues.